src/lib/server-only-api/work.ts

```diff
@@ -0,0 +1,12 @@
+import { createSupabaseServerClient } from '@/lib/supabase/server';
+import { Work } from '@/types/Work';
+
+export async function getWorks(pageIndex: number, pageSize: number) {
+    const supabaseServerClient = await createSupabaseServerClient();
+
+    return await supabaseServerClient
+        .from('works')
+        .select('*', { count: 'exact' })
```

Comment:
no need to fetch everything `*`, fetch only title & id and something else that you need?